src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,111 @@
+package duke;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.exception.WriteToStorageException;
+import duke.task.Task;
+
+
+/**
+ * Represents a list of Task objects. Contains functions to manipulate tasklist.
+ */
+public class TaskList {
+    private Storage store;
+    private List<Task> tasks;
+
+
+    /**
+     * To instantiate a TaskList, pass in Storage object.
+     *
+     * @param store  Storage Object to store Task data.
+     */
+    public TaskList(Storage store) {
+        this.store = store;
+        this.tasks = new ArrayList<>();
+        store.initialiseTasks(tasks);
+    }
+
+
+    /**
+     * Returns TaskList's Storage Object.
+     *
+     * @return TaskList's Storage Object.
+     */
+    public Storage getStore() {
+        return store;
+    }
+
+    /**
+     * Add task to TaskList
```

Comment:
Might want to change `Add` to `Adds` for to make it grammatically correct and more consistent throughout your code as I've noticed some methods use singular verbs in their description while some others use plural verbs.